src/Exceptions/ErrorInfo.php

```diff
@@ -0,0 +1,30 @@
+<?php
+namespace Ably\Exceptions;
+
+use \Exception;
+
+/**
+ * Base class for Ably exceptions, provides read-only access to its fields
+ * @property-read string $message Exception message
+ * @property-read int $code Exception code @see https://github.com/ably/ably-common/blob/master/protocol/errors.json
+ * @property-read int $statusCode HTTP error code
+ */
+class ErrorInfo extends Exception {
```

Comment:
On the other hand.. is it necessary at all, when the exception already contains code, statusCode and message? (ie. all the data from an ErrorInfo)